src/benchmarklib/table_generator.cpp

```diff
@@ -184,7 +184,7 @@ std::shared_ptr<Table> TableGenerator::generate_table(
 
       // add full chunk to table
       if (column_index == num_columns - 1) {
-        table->append_chunk(segments, allocator_chunk);
+        table->append_chunk(segments, nullptr, allocator_chunk);
```

Comment:
Admittedly spiteful comment incoming:
How is nullptr here any better than true or false? Right now, I have no idea what I am looking at.

Reply:
> Right now, I have no idea what I am looking at.

Exactly, you do not - and I will quote this during the next bool-vs-enum debate. ;)

nullptr is not categorically better than an unnamed bool, and I wouldn't claim it is. I had to use it here, because there are no real alternatives. `std::shared_ptr<MvccData>{}`, imo, doesn't read as "don't supply MvccData". Changing the parameter order of the allocator and the MvccData wouldn't help, because there are places where MvccData is supplied, but not an allocator, which would result in a call like `...->append_chunk(segments, std::nullopt, mvcc_data` - which isn't cleaner than the current state.

Two different function signature `append_chunk_with_mvcc_data()` and `append_chunk_without_mvcc_data()` would be a possibility, but then `Projection` would need a branching to decide which of these it calls. Also not better, imo.